### PR TITLE
Moved DropLoot to Lua.

### PR DESCRIPTION
**Pull request by Luckey1729**
This is the proper implementation of #762.
I accidentally had to create a new PR because I'm still not the best at using GitHub.
This allows for more flexibility and customization regarding how players drop items when they die.

### CODE REVIEW

File: data/creaturescripts/scripts/droploot.lua [hunk 1]
```diff
@@ -0,0 +1,34 @@
+function onDeath(player, corpse, killer, mostDamage, unjustified, mostDamage_unjustified)
+	if not getPlayerFlagValue(player, PlayerFlag_NotGenerateLoot) and player:getVocation() ~= VOCATION_NONE then
```
Comment (ninjalulz) [inline data/creaturescripts/scripts/droploot.lua]: Missing :getId() (```player:getVocation():getId()```)

File: data/creaturescripts/scripts/droploot.lua [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		local amulet = player:getSlotItem(CONST_SLOT_NECKLACE)
+		if amulet and amulet:getId() == ITEM_AMULETOFLOSS and player:getSkull() ~= SKULL_RED and player:getSkull() ~= SKULL_BLACK then
```
Comment (ninjalulz) [inline data/creaturescripts/scripts/droploot.lua]: Use amulet.itemid instead and you can also use isInArray for skulls.

File: data/creaturescripts/scripts/droploot.lua [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+			local isPlayer = false
+			if Player(killer) then
```
Comment (ninjalulz) [inline data/creaturescripts/scripts/droploot.lua]: change to killer:isPlayer()

File: data/creaturescripts/scripts/droploot.lua [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+				isPlayer = true
+			else
+				local master = Creature(killer):getMaster()
```
Comment (ninjalulz) [inline data/creaturescripts/scripts/droploot.lua]: change to killer:getMaster()

File: data/creaturescripts/scripts/droploot.lua [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+			if not isPlayer or isPlayer and not player:hasBlessing(6) then
+				player:removeItem(ITEM_AMULETOFLOSS, 1, -1, false)
+			end
+		else
+			local corpse = Container(corpse)
+			for i = CONST_SLOT_HEAD, CONST_SLOT_AMMO do
+				local item = player:getSlotItem(i)
+				if item then
+					if (player:getSkull() == SKULL_RED or player:getSkull() == SKULL_BLACK) or math.random(1, item:isContainer() and 100 or 1000) <= player:getLossPercent() then
```
Comment (ninjalulz) [inline data/creaturescripts/scripts/droploot.lua]: You can use isInArray here as well

**Comment by PrinterLUA** [comment]
Here you go, i've also added @ninjalulz notes:
http://pastebin.com/pApGiFR5

**Comment by gugahoa** [comment]
With those changes you can remove Player::getDropPercent too

**Comment by Luckey1729** [comment]
I'm aware of the improvements that can be made to this PR.
I'm currently very busy in my life. I'm more than welcoming to see anyone take the code I've changed and submit their own PR with the mentioned changes from my last commit. I'd rather see the code added to TFS than "have credit" for it.

If nobody does it themselves, I'll try to have the items above addressed within a few weeks.

**Comment by Luckey1729** [comment]
Let me know how this looks. I hope I squashed my previous commits fine; I'm still not the greatest at using GitHub which is one of the only reasons I haven't been doing more.

Let me know if anything else can be done with it.

File: data/creaturescripts/scripts/droploot.lua [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+		if killer:isPlayer() then
+			isPlayer = true
+		else
+			local master = killer:getMaster()
+			if master and master:isPlayer() then
+				isPlayer = true
+			end
+		end
+
+		if not isPlayer or isPlayer and not player:hasBlessing(6) then
```
Comment (marksamman) [inline data/creaturescripts/scripts/droploot.lua]: This can be changed to:
```lua
if not isPlayer or not player:hasBlessing(6) then
```

**Comment by marksamman** [comment]
Looks good to me, but I'm curious if it has been tested considering there were a couple of issues in your initial commit. If you, or anyone else, can confirm that losing an item works fine on death, I'll merge this.

**Comment by ninjalulz** [comment]
VOCATION_NONE, ITEM_AMULETOFLOSS, ITEM_BAG is not defined.

**Comment by marksamman** [comment]
Merged #1173.

**Pull request closed without merging**